### DISCUSSION TITLE
Multiple files - one config

### DISCUSSION THREAD

**Discussion by blushingbacon** [pybricks/support#1717] [Q&A]
I am a new coach for an FLL challenge team and we are using Pybricks - starting with codeblocks and working our way to python. We are interested in having different files for different missions. Am I right that the robot has to be setup/configured in each file and that there is not a way to configure the base robot in one file and then import it? I only see ways to import tasks using codeblocks.

**Comment by laurensvalk** [comment]
Here is a [tutorial](https://pybricks.com/project/spike-hub-menu/) with a video that shows how it will work.

Each of your mission programs can be made with blocks. Only the menu itself is made with Python as you'll see in the tutorial. It's only a few lines though, and a nice way to get familiar with it :slightly_smiling_face:.

--------------

> Am I right that the robot has to be setup/configured in each file and that there is not a way to configure the base robot in one file and then import it?

If you want to, you can pass the devices you set up to a generic library function, just as you would in Python. This might be fairly advanced for new team members though. Here is an [example](https://github.com/orgs/pybricks/discussions/1344#discussioncomment-7915280) of that technique.

**Reply by blushingbacon** [reply]
Thank you! I had it in my mind you could only import tasks, but it seems so obvious now that you point it out. Thank you so much for all of your work with Pybricks and your responses to this community!

**Reply by laurensvalk** [reply]
Happy to help,  have fun coding! Please tell your team that we wish them all the best :)
